packages/core/src/components/StepList/StepList.scss

```diff
@@ -0,0 +1,413 @@
+
+.ds-c-step__completed-text {
+  display: inline-block;
+  line-height: $step-button-height - 12px;
+  margin-right: $spacer-2;
+  padding-top: 6px;
+
+  &::before {
+    background-image: url('#{$image-path}/checkmark-green.svg');
+    background-size: contain;
+    box-sizing: border-box;
+    content: '';
+    display: inline-block;
+    height: 12px;
+    margin-right: $spacer-1;
+    width: 16px;
+  }
+
+  @media (min-width: $step-list-breakpoint) {
+    display: block;
+    margin-right: 0;
+  }
+}
+
+.ds-c-step__description {
+  font-size: $small-font-size;
+  margin-bottom: $spacer-1;
+
+  .ds-c-step--current &,
+  .ds-c-step--completed & {
+    margin-top: $spacer-1;
+  }
+
+  @media (min-width: $step-list-breakpoint) {
+    font-size: $base-font-size;
+  }
+}
+
+.ds-c-step__substeps {
+  list-style-type: none;
+  margin: 0;
+  padding-left: $spacer-5;
+}
+
+.ds-c-substep {
+  display: block;
+  font-size: $small-font-size;
+
+  &::before {
+    content: '\2022';
+    left: -$spacer-3;
+    position: relative;
+  }
+}
+
+.ds-c-substep__title {
+  display: inline-block;
+  margin-right: $spacer-1;
+}
+
+.ds-c-substep__edit {
+  @media (min-width: $step-list-breakpoint) {
+    position: absolute;
+    right: 0;
+  }
+}
+
+/*
+Start, Resume, and Edit links
+
+A user interacts with the steps through "Start", "Resume", and "Edit"
+links.
+
+Visually "Start" and "Resume" are buttons, but all three are implemented
+with `<a>` tags. In the React component, the links' `href` properties are
+determined by the `step` object's `href` property. One can also
+optionally pass an `onStepLinkClick` function that will cancel the default
+link behavior and call `onStepLinkClick` with `href` as a parameter for
+apps that handle routing with JavaScript.
+
+When a step has substeps and is incomplete, the `href` property of should
+be set to match the `href` of the current substep—that is, the first
+incomplete substep. The "Edit" button will only appear on substeps that
+have been completed.
+
+Style guide: patterns.step-list.buttons
+*/
+
+/*
+`<StepList>`
+
+@react-component StepList
+
+Style guide: patterns.step-list.react
+*/
+
+/*
+Step object
+
+A `step` object contains all information needed to render the step, including
+text, progress, and routing information.
+
+Step object properties:
+- **id** (`string`) - Unique string representing the step. Will be passed to onStepLinkClick as second parameter
+- **href** (`string`) - URL or partial URL that routes to the step. Will be passed to onStepLinkClick as first parameter
+- **completed** (`bool`) - Whether the step has been completed
+- **started** (`bool`) - Whether the step has been started
+- **isNextStep** (`bool`) - Whether this is the next unstarted step
+- **title** (`string`) - Text to display as the step title
+- **description** (`string`) - Additional text to display under (only rendered for top-level steps)
+
+Style guide: patterns.step-list.step-object
+*/
+
+/*
+---
+
+## Creating steps
```

Comment:
✍️ We might want to frame this as "Advanced usage: Maintaining state of complex lists" or something. This tab usually provides basic guidance that anyone implementing a pattern should follow, but while this content is useful, I'd be wary of framing it as something that applies to every implementation.

Reply:
I feel like the `hrefs` for "Resume" corresponding to the current step is important for user experience. It is not needed when there are no substeps, but I don't think it's advanced. I like the idea for changing the title to mention state management. What about "Managing list state". I get how it could be "advanced" usage, but it's really stuff that should be observed for any list

Reply:
I'm just pushing back a little bit on this. I think it's important

Reply:
Yeah, agreed that it's important that "Resume" takes you to the current step. "Managing list state" works for me. My main concern was that "Creating steps" could be potentially confusing for simple lists.